esrally/driver/runner.py

```diff
@@ -2116,6 +2117,58 @@ def __repr__(self, *args, **kwargs):
         return "delete-transform"
 
 
+class TransformStats(Runner):
+    """
+    Gather index stats for one or all transforms.
+    """
+
+    def _get(self, v, path):
+        if v is None:
+            return None
+        elif len(path) == 1:
+            return v.get(path[0])
+        else:
+            return self._get(v.get(path[0]), path[1:])
+
+    def _safe_string(self, v):
+        return str(v) if v is not None else None
+
+    async def __call__(self, es, params):
+        api_kwargs = self._default_kw_params(params)
+        transform_id = mandatory(params, "transform-id", self)
+        condition = params.get("condition")
+        response = await es.transform.get_transform_stats(transform_id=transform_id, **api_kwargs)
+        transforms = response.get("transforms", [])
+        transform_stats = transforms[0] if len(transforms) > 0 else {}
+        if condition:
+            path = mandatory(condition, "path", repr(self))
+            expected_value = mandatory(condition, "expected-value", repr(self))
+            actual_value = self._get(transform_stats, path.split("."))
+            return {
+                "weight": 1,
+                "unit": "ops",
+                "condition": {
+                    "path": path,
+                    # avoid mapping issues in the ES metrics store by always rendering values as strings
+                    "actual-value": self._safe_string(actual_value),
+                    "expected-value": self._safe_string(expected_value)
+                },
+                # currently we only support "==" as a predicate but that might change in the future
+                "success": actual_value == expected_value,
+                "transform-stats": transform_stats
```

Comment:
Is it always required to have the transform stats returned? The main concern here is around key/field explosion on the metrics store. 

Would it make sense to either not return it at all, or perhaps could it be returned only if a user passes in a specific parameter? e.g. the `BulkIndex` runner has a `detailed-results` param to determine whether or not we return detailed stats:
https://github.com/elastic/rally/blob/40334f18f92e82686c7a1b4440d53395767d2445/esrally/driver/runner.py#L462-L467

Reply:
its not needed and I wasn't aware that this is pushed to the metric store. I removed it.